src/api.tsx

```diff
@@ -427,36 +416,117 @@ export async function addMedia(name: string, type: "icon" | "photo", file: File)
 
 }
 
-
-export async function addGuideInfo(name: string, pic: File): Promise<GuideInfo> {
+function GetResourceRefOfGuidePic(pic: File) : StorageReference {
     const storage = getStorage(app);
     const storageRef = ref(storage);
     const mediaRef = ref(storageRef, 'media');
     const folderRef = ref(mediaRef, 'guides_pics');
     const resourceRef = ref(folderRef, pic.name);
 
+    return resourceRef;
+}
+
+export function isUserAdmin(user: UserInfo) : Promise<boolean> {
```

Comment:
simpler code: define the function async and return boolean:
```
export async function isUserAdmin(u:UserInfo): boolean {
if (!user._ref) {
return false;
...

Reply:
Fix in main branch